app/kuma-dp/pkg/dataplane/metrics/server.go

```diff
@@ -236,10 +267,13 @@ func selectContentType(contentTypes <-chan expfmt.Format, reqHeader http.Header)
 		}
 	}
 
-	// If no valid content type is returned by the applications,
-	// use the highest priority content type supported by the scraper.
+	// If no valid content type is returned by the target applications,
+	// negotitate content type based on Accept header of the scraper.
+	//
+	// Note: NegotiateIncludingOpenMetrics is not yet fully supported.
```

Comment:
The problem was that we were incorrectly hitting the `FmtUnknown` case due to Envoy not including the version and then we were sending text metrics as OpenMetrics which resulted in Prometheus' errors.

I think it's not that `NegotiateIncludingOpenMetrics` wouldn't work but more a question of what's the right thing to do in this case. If we hit this case, we don't have OpenMetrics so I think `Negotiate` is fine.

Reply:
Yes, as long as Envoy is present in the list of targets to scrape along with other apps, we would always have at least text/plain (Prometheus Text) format. So we can safely use `Negotiate` as it's stable and more widely implemented.

Reply:
I think we can just remove the comment, because it's nothing to do with NegotiateIncludingOpenMetrics being not yet fully supported per se it's more that we just don't want to set `Content-Type` to OpenMetrics if we don't know what content type we're returning.